ERCS/erc-7868.md

```diff
@@ -0,0 +1,117 @@
+---
+eip: 7868
+title: Verifiable Cross Rollup Links
+description: Identifiers for verifying the source of cross rollup inputs and corresponding shared settlement mechanisms.
+author: Bo Du (@notbdu), Ian Norden (@i-norden)
+discussions-to: TBD
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-01-23
+---
+
+## Abstract
+
+Interoperability between Ethereum rollups should be secured by the L1. This means that cross rollup transactions need to be verifiable at settlement time.
+
+We propose cross rollup input identifiers that contain information uniquely identifying its source or origin.
+
+This informs the underlying native or non native proof system which transactions contain cross rollup inputs that need to be verified.
+
+The identifier formats for cross rollup inputs aim to be agnostic to the rollup framework, proof system and messaging mechanics used. 
+
+We propose using a pointer to a generic log emitted on or storage key written on the origin rollup as an identifier. These pointers form "cross rollup links" which can be cryptographically linked to the execution results of communicating rollups and validated at settlement time.
+
+## Motivation
+
+Rollup stacks today cannot settle cross rollup transactions together on the L1. There is no standard way for different rollup frameworks to: 
+
+- Specify which transactions contain cross rollup inputs.
+- Verify that the source of a cross rollup input is valid at settlement time.
+
+Settlement time validation of cross rollup inputs enables rollups of different rollup frameworks to join or leave the rollup clusters they want to share fast trust-minimized communication with. This allows the formation of secure rollup economic zones on top of Ethereum.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Identifiers
+
+We propose a log and storage key identifier. Both are verifiable by cryptographically linking them to the execution results (e.g. block header) produced by communicating rollups (and the L1). The creation of either identifier creates a "cross rollup link" between the execution of the origin and the destination rollup. These cross rollup links can either be pessimistically or optimistically validated depending on the type of proof system used.
+
+The `LogIdentifier` specification below borrows heavily from Optimism's [message identifier](https://github.com/ethereum-optimism/specs/blob/main/specs/interop/messaging.md#message-identifier) format. It uniquely identifies a log emitted on the origin rollup.
```

Comment:
The optimism specs are not a permitted external origin. Because their license is compatible, you can, if you'd like, put a copy of the relevant text in the `assets` directory. Alternatively you can follow the process set out in [EIP-5757](https://eips.ethereum.org/EIPS/eip-5757) and we will consider relaxing the rules.